buildSrc/src/main/kotlin/org/cqfn/save/buildutils/PublishingConfiguration.kt

```diff
@@ -0,0 +1,147 @@
+                    .matching { it.publication == targetPublication }
+                    .configureEach {
+                        onlyIf {
+                            // main publishing CI job is executed on Linux host
+                            DefaultNativePlatform.getCurrentOperatingSystem().isLinux.apply {
+                                if (!this) {
+                                    logger.lifecycle("Publication ${(it as AbstractPublishToMaven).publication.name} is skipped on current host")
+                                }
+                            }
+                        }
+                    }
+            }
+        }
+    }
+}
+
+@Suppress("TOO_LONG_FUNCTION")
+private fun Project.configurePublications() {
+    val dokkaJar: Jar = tasks.create<Jar>("dokkaJar") {
+        group = "documentation"
+        archiveClassifier.set("javadoc")
+        from(tasks.findByName("dokkaHtml"))
+    }
+    configure<PublishingExtension> {
+        repositories {
+            mavenLocal()
+        }
+        publications.withType<MavenPublication>().forEach { publication ->
+            publication.artifact(dokkaJar)
+            publication.pom {
+                name.set(project.name)
+                description.set(project.description ?: project.name)
+                url.set("https://github.com/cqfn/save")
+                licenses {
+                    license {
+                        name.set("MIT License")
+                        url.set("http://www.opensource.org/licenses/mit-license.php")
+                        distribution.set("repo")
+                    }
+                }
+                developers {
+                    developer {
+                        id.set("petertrr")
+                        name.set("Petr Trifanov")
+                        email.set("peter.trifanov@mail.ru")
+                    }
+                    developer {
+                        id.set("akuleshov7")
+                        name.set("Andrey Kuleshov")
+                        email.set("andrewkuleshov7@gmail.com")
+                    }
+                }
+                scm {
+                    url.set("https://github.com/cqfn/save")
+                    connection.set("scm:git:git://github.com/cqfn/save.git")
+                }
+            }
+        }
+    }
+}
+
+private fun Project.configureSigning() {
+    configure<SigningExtension> {
+        useInMemoryPgpKeys(property("signingKey") as String?, property("signingPassword") as String?)
+        logger.lifecycle("The following publications are getting signed: ${extensions.getByType<PublishingExtension>().publications.map { it.name }}")
+        sign(*extensions.getByType<PublishingExtension>().publications.toTypedArray())
+    }
+}
+
+private fun Project.configureNexusPublishing() {
+    configure<NexusPublishExtension> {
+        repositories {
+            sonatype {
+                // only for users registered in Sonatype after 24 Feb 2021
```

Comment:
Need to check a url for older users, because cqfn has been registered at least a year ago